### ISSUE TITLE
package-lock.json keeps being altered.

### ISSUE THREAD

**Issue by CodingSpecies** [EddieHubCommunity/LinkFree#272]
Every single pull request made by others contains a commit that shows the package-lock.json being altered. 
However, the user and creator of the pull-request hasn't done anything or made any changes. I think it automatically updates things such as the version every time someone forks and clones, and adds changes.

Thank you.

**Comment by Zain-Muiz** [comment]
Heyy @CodingSpecies , I digged into this issue and looks like this is intended by npm. Whenever you run a `npm install` the package-lock.json file will get updated to have the newest version of dependencies. We have one possible solutions for this
1. I've found that there will be a new version of npm 5.7.1 with the new command `npm ci`, that will install from package-lock.json only ( Stack overflow ) [ This will fail the install if there are mismatch of versions in package and package-lock rather than update]

Any other possible solutions that you can add ?

**Comment by AbhijithGanesh** [comment]
Can this please be added to the local gitignore of repositories? It will be useful because changes won't be staged until the community decides to upgrade a package version

**Comment by CodingSpecies** [comment]
@Zain-Muiz I agree. 
And as said, every time someone runs npm install it updates the dependencies, changing package-lock.json.
But I am not sure how to fix this, maybe, if we add it to the gitignore as @AbhijithGanesh explained, it might help too. thanks.

**Comment by aayush89890** [comment]
```node_modules``` and ```package-lock.json``` they get created when we run npm install.
This files are not needed in the commit *we can add in the documentation.*
**We need to commit only those changes that we have made in the project.**
@CodingSpecies @Zain-Muiz @AbhijithGanesh

**Comment by AbhijithGanesh** [comment]
Yes agreed, locally add package-lock.json to .gitignore everytime you work

**Comment by aayush89890** [comment]
Yes @AbhijithGanesh correct.

**Comment by CodingSpecies** [comment]
> [ANOTHER METHOD]
> 
> Also, instead of using command ,
> 
> > git add . (which adds everything to staging area)
> 
> -One should use
> 
> > git add public/data/your-username.json (add only selected file to staging area).
> 
> And then use commit.

That is another option, but we will have to make it clear to everyone, perhaps in the readme.

**Comment by Panquesito7** [comment]
I slightly edited the issue description to make it a beginner-friendly issue. Anyone is welcome to make a pull request! 🙂

**Comment by krishguptadev** [comment]
Can I make one pr fixing the readme?
